### ISSUE TITLE
Should EdgeNGramTokenizer's DEFAULT_MAX_GRAM_SIZE be ONE?

### ISSUE THREAD

**Issue by YeonghyeonKO** [apache/lucene#13802]
### Description

From org.apache.lucene:lucene-analysis-common:9.11.1, the static variable `DEFAULT_MAX_GRAM_SIZE` of EdgeNGramTokenizer is ONE not TWO. 

Logically, the maximum n-gram size must be >= minGramSize but since many libraries(**git code**: [Elasticsearch](https://github.com/elastic/elasticsearch/blob/main/modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java#L511), [OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/main/modules/analysis-common/src/main/java/org/opensearch/analysis/common/EdgeNGramTokenizerFactory.java#L54)) use `NGramTokenizer.DEFAULT_MAX_NGRAM_SIZE` not `EdgeNGramTokenizer`'s.
Will there be any dependency problem in Solo project as a result of my suggestion?

See the below codes:

```java
public class EdgeNGramTokenizer extends NGramTokenizer {
    public static final int DEFAULT_MAX_GRAM_SIZE = 1;    /* How about changing '1' to '2'? */
    public static final int DEFAULT_MIN_GRAM_SIZE = 1;

    public EdgeNGramTokenizer(int minGram, int maxGram) {
        super(minGram, maxGram, true);
    }

    public EdgeNGramTokenizer(AttributeFactory factory, int minGram, int maxGram) {
        super(factory, minGram, maxGram, true);
    }
}

```

**Comment by jpountz** [comment]
I agree that these defaults are not practical. The only purpose of this tokenizer that I'm familiar with is speeding up prefix queries. This makes me wonder if we should have an even higher max gram size, e.g. 8 or 10.

**Comment by YeonghyeonKO** [comment]
In actual Elasticsearch settings, it is common to use values ​​of 8 or 10 as @jpountz said. Of course, people may have different preferences, but I think it is not a good idea to set both max and min values ​​to 1.
